File: app/src/release/java/de/mchllngr/androidboilerplate/base/BaseApp.java
package de.mchllngr.androidboilerplatekotlin.base;

import android.app.Application;

import de.mchllngr.androidboilerplatekotlin.util.timber.ReleaseTree;
import timber.log.Timber;

/**
 * Base-class used for initialising {@link Timber}.
 */
public class BaseApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        initTimber();
    }

    /**
     * Initialises {@link Timber} with release configuration
     */
    private void initTimber() {
        Timber.plant(new ReleaseTree());
    }
}

File: app/src/release/java/de/mchllngr/androidboilerplatekotlin/base/BaseApp.kt
package de.mchllngr.androidboilerplatekotlin.base

import android.app.Application

import de.mchllngr.androidboilerplatekotlin.util.timber.ReleaseTree
import timber.log.Timber

/**
 * Base-class used for initialising [Timber].
 */
open class BaseApp : Application() {

    override fun onCreate() {
        super.onCreate()

        initTimber()
    }

    /**
     * Initialises [Timber] with release configuration
     */
    private fun initTimber() {
        Timber.plant(ReleaseTree())
    }
}
